validation/validation.go

```diff
@@ -1,22 +1,179 @@
 package validation
 
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+
+	"github.com/helm/helm/chart"
+	"github.com/helm/helm/log"
+	"github.com/helm/helm/manifest"
+	"gopkg.in/yaml.v2"
+)
+
+// ChartValidation represents a specific instance of validation against a specific directory
+type ChartValidation struct {
+	Path         string
+	Validations  []*Validation
+	Chartfile    *chart.Chartfile
+	Manifests    []*manifest.Manifest
+	ErrorCount   int
+	WarningCount int
+}
+
+const (
+	warningLevel = 1
+	errorLevel   = 2
+)
+
 // Validation - hold messages related to validation of something
 type Validation struct {
-	Errors   []string
-	Warnings []string
+	children  []*Validation
+	path      string
+	validator validator
+	Message   string
+	level     int
+}
+
+//ChartYamlPath - path to Chart.yaml
+func (v *Validation) ChartYamlPath() string {
+	return filepath.Join(v.path, "Chart.yaml")
+}
+
+//ChartManifestsPath - path to Manifests directory
+func (v *Validation) ChartManifestsPath() string {
+	return filepath.Join(v.path, "manifests")
+}
+
+func (v *Validation) Chartfile() (*chart.Chartfile, error) {
+	var y *chart.Chartfile
+	b, err := ioutil.ReadFile(v.ChartYamlPath())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err = yaml.Unmarshal(b, &y); err != nil {
+		return nil, err
+	}
+
+	return y, nil
 }
 
-// AddWarning - add warning to validation
-func (v *Validation) AddWarning(w string) {
-	v.Warnings = append(v.Warnings, w)
+type validator func(path string, v *Validation) (result bool)
```

Comment:
Can you export this? It'll be confusing in the godoc on `AddError` to see a `validator` type, but not know what it is